src/pub.c

```diff
@@ -1563,11 +1566,12 @@ DPS_Status DPS_Publish(DPS_Publication* pub, const uint8_t* payload, size_t len,
         if (pub->historyCap > 1) {
             DPS_ERRPRINT("History depth > 1 only valid for non-retained publications\n");
             ret = DPS_ERR_INVALID;
-            goto Exit;
+            goto Unlock;
         }
         clone->flags |= PUB_FLAG_RETAINED;
     }
     ++clone->sequenceNum;
+    DPS_UnlockNode(node);
```

Comment:
It would be cleaner and remove the need for separate Lock and Exit labels to just call DPS_LockNode() in the one case below before calling goto Exit.

Reply:
I'll fix that up